ttnn/cpp/ttnn/operations/data_movement/slice/slice_pybind.hpp

```diff
@@ -46,17 +47,17 @@ void bind_slice(py::module& module) {
         ttnn::pybind_overload_t{
             [] (const OperationType& self,
                 const ttnn::Tensor& input_tensor,
-                const tt::tt_metal::Array4D & slice_start,
-                const tt::tt_metal::Array4D & slice_end,
-                const std::optional<tt::tt_metal::Array4D> &step,
+                const std::vector<int> &slice_start,
+                const std::vector<int> &slice_end,
+                const std::optional<std::vector<int>> &step,
                 const std::optional<ttnn::MemoryConfig>& memory_config,
                 uint8_t queue_id) {
-                    return self(queue_id, input_tensor, slice_start, slice_end, step, memory_config);
+                    return self(queue_id, input_tensor, slice_start, slice_end, step.value_or(vector<int>(slice_end.size(), 1)), memory_config);
                 },
                 py::arg("input_tensor"),
                 py::arg("slice_start"),
                 py::arg("slice_end"),
-                py::arg("step") = std::nullopt,
+                py::arg("step") = std::nullopt, // should consider a better default value
```

Comment:
any ideas?

Reply:
I originally had all 1s, but this is actually bad because step size needs to match begins/ends which can now be 4D. As a result the best choice is either nullopt or forcing the user to always input the step as well.